# Kjøre modellen lokalt
### Laste inn modellen fra Hugging Face

In [16]:
# Load model directly
from transformers import AutoModelForCausalLM, pipeline

model = AutoModelForCausalLM.from_pretrained("microsoft/phi-2",  trust_remote_code=True)


Loading checkpoint shards: 100%|██████████| 2/2 [00:09<00:00,  4.74s/it]


In [17]:
from transformers import AutoTokenizer
tokenizer = AutoTokenizer.from_pretrained("microsoft/phi-2", trust_remote_code=True)


Special tokens have been added in the vocabulary, make sure the associated word embeddings are fine-tuned or trained.


In [18]:
# Use a pipeline as a high-level helper
from transformers import pipeline

pipe = pipeline("text-generation", model=model, tokenizer=tokenizer)

In [40]:
input_to_model = "A skier slides down a frictionless slope of height 40m and length 80m. What's the skier's speed at the bottom?"
res = pipe(input_to_model, max_new_tokens=500)
print(res[0]['generated_text'])

[{'generated_text': "A skier slides down a frictionless slope of height 40m and length 80m. What's the skier's speed at the bottom?\n\nSolution:\nThe skier's speed at the bottom is the same as the speed of a freely falling object with a height of 40m. Using the formula v^2 = u^2 + 2gh, where u is the initial velocity (which is 0), g is the acceleration due to gravity (9.8 m/s^2), and h is the height (40m), we get v = sqrt(2gh) = sqrt(2*9.8*40) = 19.8 m/s.\n\nFollow-up Exercise 1:\nWhat if the slope is not frictionless? How would the answer change?\n\nSolution:\nIf the slope is not frictionless, then the skier will experience a frictional force that will slow them down. The amount of frictional force depends on the coefficient of friction between the skis and the slope, as well as the normal force exerted by the slope on the skis. The normal force is equal to the weight of the skier, which is mg, where m is the mass of the skier and g is the acceleration due to gravity. The frictional f

#### max_new_tokens=50  -  43s

A skier slides down a frictionless slope of height 40m and length 80m. What's the skier's speed at the bottom?

Solution:
The skier's speed at the bottom is the same as the speed of a freely falling object with a height of 40m. Using the formula v^2 = u^2 + 2gh, where u is the initial

#### max_new_tokens=100  -  1m 3s

A skier slides down a frictionless slope of height 40m and length 80m. What's the skier's speed at the bottom?

Solution:
The skier's speed at the bottom is the same as the speed of a freely falling object with a height of 40m. Using the formula v^2 = u^2 + 2gh, where u is the initial velocity (which is 0), g is the acceleration due to gravity (9.8 m/s^2), and h is the height (40m), we get v = sqrt(2gh) = sqrt(2*9.

### max_new_tokens=150  -  1m 25s

A skier slides down a frictionless slope of height 40m and length 80m. What's the skier's speed at the bottom?

Solution:
The skier's speed at the bottom is the same as the speed of a freely falling object with a height of 40m. Using the formula v^2 = u^2 + 2gh, where u is the initial velocity (which is 0), g is the acceleration due to gravity (9.8 m/s^2), and h is the height (40m), we get v = sqrt(2gh) = sqrt(2*9.8*40) = 19.8 m/s.

Follow-up Exercise 1:
What if the slope is not frictionless? How would the answer change?

Solution:
If the slope is not frictionless, then

### max_new_tokens=500  -  4m 2s

A skier slides down a frictionless slope of height 40m and length 80m. What's the skier's speed at the bottom?

Solution:
The skier's speed at the bottom is the same as the speed of a freely falling object with a height of 40m. Using the formula v^2 = u^2 + 2gh, where u is the initial velocity (which is 0), g is the acceleration due to gravity (9.8 m/s^2), and h is the height (40m), we get v = sqrt(2gh) = sqrt(2*9.8*40) = 19.8 m/s.

Follow-up Exercise 1:
What if the slope is not frictionless? How would the answer change?

Solution:
If the slope is not frictionless, then the skier will experience a frictional force that will slow them down. The amount of frictional force depends on the coefficient of friction between the skis and the slope, as well as the normal force exerted by the slope on the skis. The normal force is equal to the weight of the skier, which is mg, where m is the mass of the skier and g is the acceleration due to gravity. The frictional force is given by f = uN, where u is the coefficient of friction and N is the normal force. The net force on the skier is then F = ma = mg - uN. Using Newton's second law, we can write v^2 = u^2 + 2gh - 2uN/m, where m is the mass of the skier. Solving for v, we get v = sqrt(u^2 + 2gh - 2uN/m).

Follow-up Exercise 2:
What if the skier is wearing a helmet that reduces the air resistance? How would the answer change?

Solution:
If the skier is wearing a helmet that reduces the air resistance, then the frictional force will be less, and the skier will go faster. The amount of reduction in air resistance depends on the shape and size of the helmet, as well as the speed of the skier. The net force on the skier is then F = ma = mg - uN - kv^2/2, where k is a constant that depends on the shape and size of the helmet. Solving for v, we get v = sqrt(u^2 + 2gh - 2uN/m - k

### Norsk fungerer svært dårlig

In [42]:
input_to_model = "Simen har to bonsai trær som er av type ficus. Han har også en orkide som heter Ola. Hvor mange trær har Simen?"
res = pipe(input_to_model, max_new_tokens=200)
print(res[0]['generated_text'])

Simen har to bonsai trær som er av type ficus. Han har også en orkide som heter Ola. Hvor mange trær har Simen?

Simen: Han har en trær som er en ficus. Han har en orkide som er en ficus. Han har en trær som er en ficus.

Lena: Han har en trær som er en ficus. Han har en orkide som er en ficus. Han har en trær som er en ficus.

Simen: Han har en trær som er en ficus. Han har en orkide som er en ficus. Han har en trær som er en ficus.

Lena: Han har en trær som er en ficus. Han har en orkide som er en ficus. Han har en trær som er en ficus.

Simen: Han har en trær som er en ficus. Han har en orkide som er en ficus. Han har en trær som


### max_new_tokens=2000  -  1m 28s

Simen har to bonsai trær som er av type ficus. Han har også en orkide som heter Ola. Hvor mange trær har Simen?

Simen: Han har en trær som er en ficus. Han har en orkide som er en ficus. Han har en trær som er en ficus.

Lena: Han har en trær som er en ficus. Han har en orkide som er en ficus. Han har en trær som er en ficus.

Simen: Han har en trær som er en ficus. Han har en orkide som er en ficus. Han har en trær som er en ficus.

Lena: Han har en trær som er en ficus. Han har en orkide som er en ficus. Han har en trær som er en ficus.

Simen: Han har en trær som er en ficus. Han har en orkide som er en ficus. Han har en trær som

In [43]:
input_to_model = "Simen has two bonsai trees of the ficus type. He also has an orchid named Ola. How many trees does Simen have?"
res = pipe(input_to_model, max_new_tokens=200)
print(res[0]['generated_text'])

Simen has two bonsai trees of the ficus type. He also has an orchid named Ola. How many trees does Simen have?

Solution:
To find the total number of trees Simen has, we need to add the number of bonsai trees and the number of orchids.

Given:
Number of bonsai trees = 2
Number of orchids = 1

To find the total number of trees, we add the number of bonsai trees and the number of orchids:
Total number of trees = Number of bonsai trees + Number of orchids
Total number of trees = 2 + 1
Total number of trees = 3

Therefore, Simen has a total of 3 trees.



Question 1: 

A store sells apples for $0.50 each and oranges for $0.75 each. If John buys 4 apples and 3 oranges, how much does he spend in total?

Solution:

To find the total amount John spends, we need to calculate the cost of the apples and


### max_new_tokens=2000  -  1m 25s

Simen has two bonsai trees of the ficus type. He also has an orchid named Ola. How many trees does Simen have?

Solution:
To find the total number of trees Simen has, we need to add the number of bonsai trees and the number of orchids.

Given:
Number of bonsai trees = 2
Number of orchids = 1

To find the total number of trees, we add the number of bonsai trees and the number of orchids:
Total number of trees = Number of bonsai trees + Number of orchids
Total number of trees = 2 + 1
Total number of trees = 3

Therefore, Simen has a total of 3 trees.



Question 1: 

A store sells apples for $0.50 each and oranges for $0.75 each. If John buys 4 apples and 3 oranges, how much does he spend in total?

Solution:

To find the total amount John spends, we need to calculate the cost of the apples and